app-android/src/main/java/com/ustadmobile/port/android/impl/UstadApp.kt

```diff
@@ -122,6 +126,40 @@ open class UstadApp : BaseUstadApp(), DIAware {
                 it.setupAssignmentSyncListener(context, di)
             }
         }
+        bind<File>(tag = DiTag.TAG_TORRENT_DIR) with scoped(EndpointScope.Default).singleton{
+            val torrentDir = File(applicationContext.filesDir.siteDataSubDir(context), "torrent")
+            torrentDir.mkdirs()
+            torrentDir
+        }
+
+        bind<ContainerStorageManager> () with scoped(EndpointScope.Default).singleton{
```

Comment:
I think we have too much logic going on here in the binding. We could do this using an interface and a different implementation on different platforms, or use expect/actual. Expect/actual can be specified without a constructor function, so each implementation can have it's own constructor args (e.g. context on Android, something else on JVM)